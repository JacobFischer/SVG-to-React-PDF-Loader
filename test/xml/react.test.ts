import React from "react";
import renderer from "react-test-renderer";
import { loadParsed, LONG_TIMEOUT } from "./webpack-compile";

describe("React output", () => {
    test(
        "Renders",
        async () => {
            const result = await loadParsed("circle.svg");
            expect(
                renderer.create(React.createElement(result.default)).toJSON(),
            ).toMatchSnapshot();
        },
        LONG_TIMEOUT,
    );

    test(
        "Accepts a getComponent prop",
        async () => {
            const result = await loadParsed("note.xml");
            const getComponent = (tag: string) => {
                /* eslint-disable @typescript-eslint/ban-types */
                switch (tag) {
                    case "note":
                        return (props: {}) =>
                            React.createElement("section", props);
                    case "heading":
                        return (props: {}) => React.createElement("h1", props);
                    default:
                        return (props: {}) => React.createElement("p", props);
                }
                /* eslint-enable @typescript-eslint/ban-types */
            };
            expect(
                renderer
                    .create(
                        React.createElement(result.default, { getComponent }),
                    )
                    .toJSON(),
            ).toMatchSnapshot();
        },
        LONG_TIMEOUT,
    );
});
