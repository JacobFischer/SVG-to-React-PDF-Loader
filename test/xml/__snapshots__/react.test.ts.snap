// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React output Accepts a getComponent prop 1`] = `
<section
  getComponent={[Function]}
>
  <p>
    Tove
  </p>
  <p>
    Jani
  </p>
  <h1>
    Reminder
  </h1>
  <p>
    Don't forget 
    <p>
      me
    </p>
     this weekend!
  </p>
</section>
`;

exports[`React output Renders 1`] = `
<svg
  height="100"
  width="100"
>
  <circle
    cx="50"
    cy="50"
    fill="red"
    r="40"
    stroke="black"
    stroke-width="3"
  />
</svg>
`;
